app/controllers/dashboard_controller.rb

```diff
@@ -0,0 +1,105 @@
+require "presenters/form_presenter"
+require "presenters/index_presenter"
+require "presenters/show_presenter"
+
+class DashboardController < ApplicationController
+  def self.dashboard_controller_for(resource_name)
+    @@resource_name = resource_name
+  end
+
+  before_action :set_resource, only: [:show, :edit, :update, :destroy]
```

Comment:
Let's get rid of this `before_action`, and be more explicit about setting the resource in each action.